test/integration/dialects/mssql/regressions.test.js

```diff
@@ -108,4 +108,33 @@ if (dialect.match(/^mssql/)) {
       });
     });
   });
+
+  it('saves value bigger than 2147483647, #11245', function() {
+    const BigIntTable =  this.sequelize.define('BigIntTable', {
+      business_id: {
+        type: Sequelize.BIGINT,
+        allowNull: false
+      }
+    }, {
+      freezeTableName: true
+    });
+
+    const bigIntValue = 2147483648;
+
+    return BigIntTable.sync({ force: true })
+      .then(() => {
+        return BigIntTable.create({
+          business_id: bigIntValue
+        });
+      })
+      .then(() => {
+        return BigIntTable.findAll({
+          where: {}
+        });
+      })
+      .then(records => {
+        expect(records).to.have.lengthOf(1);
+        expect(Number(records[0].business_id)).to.equals(bigIntValue);
+      });
```

Comment:
This can be simplified a bit. Please change this part:

```js
.then(() => {
  return BigIntTable.findAll({
    where: {}
  });
})
.then(records => {
  expect(records).to.have.lengthOf(1);
  expect(Number(records[0].business_id)).to.equals(bigIntValue);
});
```

to:

```js
.then(() => {
  return BigIntTable.findOne();
.then(record => {
  expect(Number(record.business_id)).to.equal(bigIntValue);
});
```

Reply:
Good point. Updated code following your recommendation. I made an arrow function `() => BigIntTable.findOne()` a one liner. I hope that will simplify the code even further.